rotkehlchen/chain/evm/decoding/aura_finance/utils.py

```diff
@@ -0,0 +1,153 @@
+
+    May raise:
+    - NotERC20Conformant
+    """
+    chain_id = vault['chain_id']
+    aura_pool_token = get_or_create_evm_token(
+        userdb=database,
+        evm_address=vault['aura_pool_token'],
+        chain_id=chain_id,
+        encounter=TokenEncounterInfo(
+            description=f'Querying {chain_id.to_name()} aura finance balances',
+            should_notify=False,
+        ),
+        underlying_tokens=[
+            UnderlyingToken(
+                address=get_or_create_evm_token(
+                    userdb=database,
+                    chain_id=chain_id,
+                    evm_address=vault['underlying_bpt_token'],
+                    encounter=TokenEncounterInfo(
+                        description=f'Querying {chain_id.to_name()} aura finance balances',
+                        should_notify=False,
+                    ),
+                ).evm_address,
+                token_kind=EvmTokenKind.ERC20,
+                weight=ONE,
+            ),
+        ],
+        protocol=CPT_AURA_FINANCE,
+    )
+
+    if aura_pool_token.protocol != CPT_AURA_FINANCE:
+        log.debug(
+            f'Updating protocol for {chain_id.to_name()} aura finance '
+            f'asset {aura_pool_token}',
+        )
+        GlobalDBHandler.set_token_protocol_if_missing(
+            token=aura_pool_token,
+            new_protocol=CPT_AURA_FINANCE,
+        )
+
+
+def _query_aura_pools(evm_inquirer: 'EvmNodeInquirer') -> list[dict[str, Any]] | None:
+    """Query Aura pool data from the chain, returning None if the query fails."""
+    booster_contract = EvmContract(
+        address=CHAIN_ID_TO_BOOSTER_ADDRESSES[evm_inquirer.chain_id],
+        abi=AURA_BOOSTER_ABI,
+        deployed_block=0,  # unused in this context
+    )
+    try:
+        latest_pools_count = evm_inquirer.call_contract(
+            abi=booster_contract.abi,
+            method_name='poolLength',
+            contract_address=booster_contract.address,
+        )
+        results = evm_inquirer.multicall(
+            calls=[
+                (
+                    booster_contract.address,
+                    booster_contract.encode(
+                        method_name='poolInfo',
+                        arguments=[idx],
+                    ),
+                ) for idx in range(latest_pools_count)
+            ],
+        )
+    except RemoteError as e:
+        log.error(f'Failed to retrieve aura pools due to {e!s}')
+        return None
+
+    if len(results) == 0:
+        log.debug(f'No aura pools found on {evm_inquirer.chain_name}')
+        return None
+
+    pools = []
+    for idx, result in enumerate(results):
+        try:
+            pool_info = booster_contract.decode(
+                result=result,
+                method_name='poolInfo',
+                arguments=[idx],
+            )
+            pools.append({
+                'chain_id': evm_inquirer.chain_id,
+                'aura_pool_token': to_checksum_address(pool_info[3]),
```

Comment:
don't use `to_checksum_address`. We have `deserialize_evm_address` which raises a proper `DeserializationError` which you need to catch here